cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -202,13 +202,17 @@ public void run() {
         LOG.info("Skipping best proposal precomputing because load monitor does not have enough snapshots.");
         // Check in 30 seconds to see if the load monitor has sufficient number of snapshots.
         sleepTime = 30000L;
-      } else if (!validCachedProposal()) {
-        LOG.debug("Invalidated cache. Cached model generation: {}, current model generation: {}",
+      } else if (!validCachedProposal() || _bestProposal.isExcludingRequestedTopics()) {
```

Comment:
`_bestProposal.isExcludingRequestedTopics()` seems always be `false` since `_bestProposal` is only updated in `computeBestProposal()`.